servers/rendering/rasterizer_rd/rasterizer_scene_rd.cpp

```diff
@@ -162,37 +162,1572 @@ void RasterizerSceneRD::_create_reflection_fast_filter(ReflectionData &rd, bool
+		//configuration changed, erase
+		_sdfgi_erase(rb);
+	}
+
+	SDFGI *sdfgi = rb->sdfgi;
+	if (sdfgi == nullptr) {
+		//re-create
+		rb->sdfgi = memnew(SDFGI);
+		sdfgi = rb->sdfgi;
+		sdfgi->cascade_mode = env->sdfgi_cascades;
+		sdfgi->min_cell_size = env->sdfgi_min_cell_size;
+		sdfgi->uses_occlusion = env->sdfgi_use_occlusion;
+		sdfgi->y_scale_mode = env->sdfgi_y_scale;
+		static const float y_scale[3] = { 1.0, 1.5, 2.0 };
+		sdfgi->y_mult = y_scale[sdfgi->y_scale_mode];
+		static const int cascasde_size[3] = { 4, 6, 8 };
+		sdfgi->cascades.resize(cascasde_size[sdfgi->cascade_mode]);
+		sdfgi->probe_axis_count = SDFGI::PROBE_DIVISOR + 1;
+		sdfgi->solid_cell_ratio = sdfgi_solid_cell_ratio;
+		sdfgi->solid_cell_count = uint32_t(float(sdfgi->cascade_size * sdfgi->cascade_size * sdfgi->cascade_size) * sdfgi->solid_cell_ratio);
+
+		float base_cell_size = sdfgi->min_cell_size;
+
+		RD::TextureFormat tf_sdf;
+		tf_sdf.format = RD::DATA_FORMAT_R8_UNORM;
+		tf_sdf.width = sdfgi->cascade_size; // Always 64x64
+		tf_sdf.height = sdfgi->cascade_size;
+		tf_sdf.depth = sdfgi->cascade_size;
+		tf_sdf.type = RD::TEXTURE_TYPE_3D;
+		tf_sdf.usage_bits = RD::TEXTURE_USAGE_SAMPLING_BIT | RD::TEXTURE_USAGE_STORAGE_BIT | RD::TEXTURE_USAGE_CAN_COPY_TO_BIT | RD::TEXTURE_USAGE_CAN_COPY_FROM_BIT;
+
+		{
+			RD::TextureFormat tf_render = tf_sdf;
+			tf_render.format = RD::DATA_FORMAT_R16_UINT;
+			sdfgi->render_albedo = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+			tf_render.format = RD::DATA_FORMAT_R32_UINT;
+			sdfgi->render_emission = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+			sdfgi->render_emission_aniso = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+
+			tf_render.format = RD::DATA_FORMAT_R8_UNORM; //at least its easy to visualize
+
+			for (int i = 0; i < 8; i++) {
+				sdfgi->render_occlusion[i] = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+			}
+
+			tf_render.format = RD::DATA_FORMAT_R32_UINT;
+			sdfgi->render_geom_facing = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+
+			tf_render.format = RD::DATA_FORMAT_R8G8B8A8_UINT;
+			sdfgi->render_sdf[0] = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+			sdfgi->render_sdf[1] = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+
+			tf_render.width /= 2;
+			tf_render.height /= 2;
+			tf_render.depth /= 2;
+
+			sdfgi->render_sdf_half[0] = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+			sdfgi->render_sdf_half[1] = RD::get_singleton()->texture_create(tf_render, RD::TextureView());
+		}
+
+		RD::TextureFormat tf_occlusion = tf_sdf;
+		tf_occlusion.format = RD::DATA_FORMAT_R16_UINT;
+		tf_occlusion.shareable_formats.push_back(RD::DATA_FORMAT_R16_UINT);
+		tf_occlusion.shareable_formats.push_back(RD::DATA_FORMAT_R4G4B4A4_UNORM_PACK16);
```

Comment:
Fails on macOS:
```
[mvk-error] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R4G4B4A4_UNORM_PACK16 is not supported on this device.
```

Reply:
Supported formats for the reference:

<details>
  <summary>Click to expand</summary>

```
	Formats: count = 50
		SurfaceFormat[0]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
		SurfaceFormat[1]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
		SurfaceFormat[2]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
		SurfaceFormat[3]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
		SurfaceFormat[4]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
		SurfaceFormat[5]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
		SurfaceFormat[6]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
		SurfaceFormat[7]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
		SurfaceFormat[8]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
		SurfaceFormat[9]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
		SurfaceFormat[10]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
		SurfaceFormat[11]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
		SurfaceFormat[12]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
		SurfaceFormat[13]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
		SurfaceFormat[14]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
		SurfaceFormat[15]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
		SurfaceFormat[16]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
		SurfaceFormat[17]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
		SurfaceFormat[18]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
		SurfaceFormat[19]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
		SurfaceFormat[20]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
		SurfaceFormat[21]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
		SurfaceFormat[22]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
		SurfaceFormat[23]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
		SurfaceFormat[24]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
		SurfaceFormat[25]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
		SurfaceFormat[26]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
		SurfaceFormat[27]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
		SurfaceFormat[28]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
		SurfaceFormat[29]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
		SurfaceFormat[30]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
		SurfaceFormat[31]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
		SurfaceFormat[32]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
		SurfaceFormat[33]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
		SurfaceFormat[34]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
		SurfaceFormat[35]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
		SurfaceFormat[36]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
		SurfaceFormat[37]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
		SurfaceFormat[38]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
		SurfaceFormat[39]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
		SurfaceFormat[40]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
		SurfaceFormat[41]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
		SurfaceFormat[42]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
		SurfaceFormat[43]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
		SurfaceFormat[44]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
		SurfaceFormat[45]:
			format = FORMAT_B8G8R8A8_UNORM
			colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
		SurfaceFormat[46]:
			format = FORMAT_B8G8R8A8_SRGB
			colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
		SurfaceFormat[47]:
			format = FORMAT_R16G16B16A16_SFLOAT
			colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
		SurfaceFormat[48]:
			format = FORMAT_A2B10G10R10_UNORM_PACK32
			colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
		SurfaceFormat[49]:
			format = FORMAT_A2R10G10B10_UNORM_PACK32
			colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
```

</details>